### PR TITLE
Add multiline messages support and add :CodiExpand

### LINKED ISSUE #27: breaking up larger outputs.
Is there a way for Codi to break a longer list up onto multiple lines? I've been using Codi with pwntools for exploitation (It's actually working fairly well!) and commonly some of the lists have 20 or so strings and it's off the screen. Thoughts?

### PR DESCRIPTION
**Pull request by AnonymusRaccoon**
This allows users to open a popup with multi-line output for a single line of code. This is particularly useful for long, pretty-printed objects as you often only see the closing bracket (as seen on the picture below)

Fix #27 (with floating windows instead of folds)

image

### CODE REVIEW

File: autoload/codi.vim [hunk 1]
```diff
@@ -625,11 +631,16 @@ function! s:preprocess_and_parse(output, interpreter, num_lines)
...
+          call add(outtbl, out_lines)
           let taken = ''
+          let out_lines = []
         else
           let passed_first = 1
         endif
       else
+        if passed_first
+          call add(out_lines, l)
+        endif
```
Comment (metakirby5) [inline autoload/codi.vim:643]: Nit: Could we move this into the below condition?

```vim
if passed_first
  call add(out_lines, l)
  if l =~? '^\S'
    let taken = l
  endif
endif
```

File: autoload/codi.vim [hunk 2]
```diff
@@ -879,8 +891,44 @@ function _G.codi_select(interpreters)
...
+  let lineidx = line(".") - 1
+  if lineidx >= len(s:results)
+    return
+  endif
+  let lines = s:results[lineidx]
+  if len(lines) ==  0
+    return
+  endif
+
+  if has("nvim")
```
Comment (metakirby5) [inline autoload/codi.vim:928]: Since we do the `has("nvim")` check in `plugin/codi.vim`, can we remove it from here?

File: doc/codi.txt [hunk 3]
```diff
@@ -175,6 +175,13 @@ USAGE                                                             *codi-usage*
              Opens a popup menu to select the filetype and open a new Codi
              scratch buffer (see :CodiNew).
 
+             Only available on neovim.
+
+                                                                 *:CodiExpand*
+:CodiExpand
```
Comment (metakirby5) [inline doc/codi.txt:181]: Can we add this to the `Usage` section of the readme?

**Review by metakirby5 — CHANGES_REQUESTED** [review]
This is very cool! It would be great if you could add a screenshot to the readme, perhaps under the usage section.

File: autoload/codi.vim (hunk 2 above)
Comment (AnonymusRaccoon) [reply autoload/codi.vim:928]: We could, but I kept it here to help someone who could want to implement a vim version. I can remove it if you want.

File: doc/codi.txt [hunk 4]
```diff
@@ -25,6 +25,9 @@ USAGE.............................................................|codi-usage|
     :Codi!............................................................|:Codi!|
     :Codi!! {filetype}...............................................|:Codi!!|
     :CodiUpdate..................................................|:CodiUpdate|
+    :CodiNew........................................................|:CodiNew|
+    :CodiSelect.................................................|:CodiiSelect|
+    :CodiExpand..................................................|:CodiExpand|
```
Comment (metakirby5) [inline doc/codi.txt:30]: Thanks for adding these in!

File: autoload/codi.vim (hunk 2 above)
Comment (metakirby5) [reply autoload/codi.vim:928]: I see, keeping it here sounds fine!

**Review by metakirby5 — APPROVED** [review]

**Comment by metakirby5** [comment]
Looks great, thank you for another nice contribution!

**Pull request merged**